### PR TITLE
Data: Improve `hasResolvingSelectors` redux metadata selector

### PR DESCRIPTION
**Pull request by tyxla**
## What?
This PR adds a couple of improvements to the `hasResolvingSelectors` selector to address the feedback provided in https://github.com/WordPress/gutenberg/pull/50222#discussion_r1200516933.

## Why?
To address some feedback and use the opportunity to improve and better document the selector.

## How?
We're essentially doing a couple of improvements:
* Removing an unnecessary array cloning, since `Object.values()` already returns an array and cloning is unnecessary since there's no way to mutate it.
* Documenting the internal `EquivalentKeyMap` property usage.

## Testing Instructions
* Verify site editor still loads correctly.
* Verify all checks are green.

### Testing Instructions for Keyboard
None

## Screenshots or screencast 
None

### CODE REVIEW

File: packages/data/src/redux-store/metadata/selectors.js [hunk 1]
```diff
@@ -140,7 +140,14 @@ export function getCachedResolvers( state ) {
  * @return {boolean} True if one or more selectors are resolving, false otherwise.
  */
 export function hasResolvingSelectors( state ) {
-	return [ ...Object.values( state ) ].some( ( selectorState ) =>
+	return Object.values( state ).some( ( selectorState ) =>
+		/**
+		 * This uses the internal `_map` property of `EquivalentKeyMap` for
+		 * optimization purposes, since the `EquivalentKeyMap` implementation
+		 * does not support a `.some()` implementation.
```
Comment (mcsf) [inline packages/data/src/redux-store/metadata/selectors.js]:
```suggestion
		 * This uses the internal `_map` property of `EquivalentKeyMap` for
		 * optimization purposes, since the `EquivalentKeyMap` implementation
		 * does not support a `.values()` implementation.
```

Unless I misunderstood, I believe you have a typo here. By analogy with the standard `Map`, we expect the prototype to contain `values()` but not `some()` (besides, what would this `some()` iterate over? Just the values? Keys? Keys and values?).

**Review by mcsf — APPROVED** [review]
I'm optimistically marking this as green, but I flagged a thing in the review. :)

Thanks for addressing my feedback!

File: packages/data/src/redux-store/metadata/selectors.js (hunk 1 above)
Comment (tyxla) [reply packages/data/src/redux-store/metadata/selectors.js]: ~Oh well, `.some()` is what we intend to use here and it is what we actually use from `Array.prototype` once we convert the map to an array. We could rephrase this to use `.values()` or `.entries()` but I think since `.some()` is what we are practicaly using, it makes most sense to mention that in the documentation. We could mention `.values()` but it wouldn't immediately be clear as to where we'd need `.values()` since we never call `Map.prototype.values()`. Does this make sense?~

I think you're right, will update. We do need only the values anyway.

Comment (tyxla) [reply packages/data/src/redux-store/metadata/selectors.js]: Hmm, I actually think I misunderstood you. Let me correct this.

Comment (tyxla) [reply packages/data/src/redux-store/metadata/selectors.js]: Updated. Thanks for the feedback!
